src/main/java/com/openlattice/datastore/edm/controllers/EdmController.java

```diff
@@ -787,6 +792,10 @@ public Void deleteEntityType( @PathVariable( ID ) UUID entityTypeId ) {
     @ResponseStatus( HttpStatus.OK )
     public UUID createPropertyType( @RequestBody PropertyType propertyType ) {
         ensureAdminAccess();
+
+        if ( EdmPrimitiveTypeKind.Binary == propertyType.getDatatype() && propertyType.getPostgresIndexType() != IndexType.NONE ){
+            throw new BadRequestException( "Indexes are not allowed on Binary property types" );
```

Comment:
I think it's probably better to include this in the PropertyType constructor in case property types are ever created through some path other than this endpoint

Reply:
That's what I was thinking initially (and did that w/out the exception) but since the PropertyType constructor is a JSONCreator / used for deserialization, I think if we create a Binary property type and then explicitly update its indexType, then later when its deserialized elsewhere we'll get explosions.

Reply:
True, although including it in the constructor would prevent it from being written into postgres (so deserialization should **theoretically** never happen) and would make it more visible if there's some service trying to create an index on binary data / expecting that index to exist. But, also true about exploding things if we get into a weird state

Reply:
Hmm. I may have misunderstood part of this ticket 😛 
Can you take a look at this: https://github.com/openlattice/conductor-client/pull/463 ?
My thought was that we would potentially want to intentionally set the index type on binary data later on to something other than `NONE`

Reply:
Well all the binary data is actually stored in S3 and not in postgres, so there's not really anything to create an index on

Reply:
Ahh, okay. I'll just get rid of this portion then!

Reply:
Err, above referenced portion